frontend/templates/troubleshooting/toc.tsx

```diff
@@ -105,45 +106,36 @@ export function MobileTOC({
          return;
       }
 
-      // There is no easy way to detect a scroll is finished
-      // if we are smooth scrolling, a set state will re-render
-      // and cause the scroll to stop abruptly
-      // so we have to wait until the scroll is stable
-      const waitUntilScrollStable = () => {
-         return new Promise((resolve) => {
-            let lastScrollY = window.scrollY;
-
-            const interval = setInterval(() => {
-               if (lastScrollY === window.scrollY) {
-                  clearInterval(interval);
-                  resolve(window.scrollY);
-               }
-               lastScrollY = window.scrollY;
-            }, 20);
-         });
-      };
-
       const onScroll = () => {
-         waitUntilScrollStable().then(() => {
-            const scrolledIntoContent =
-               (contentRef.current?.offsetTop || 0) <= window.scrollY;
+         const scrolledIntoContent =
+            (contentRef.current?.offsetTop || 0) <= window.scrollY;
 
-            setShowMobileTOC((_prev) => {
-               return scrolledIntoContent;
-            });
+         setShowMobileTOC((_prev) => {
+            return scrolledIntoContent;
          });
       };
 
       window.addEventListener('scroll', onScroll);
       return () => window.removeEventListener('scroll', onScroll);
    }, [contentRef]);
 
+   const isVisible = showMobileTOC && activeItem;
+
    return (
-      <Collapse in={Boolean(showMobileTOC && activeItem)} unmountOnExit={true}>
-         <Flex {...props}>
-            <MobileTOCMenu activeItem={activeItem} items={items} />
-         </Flex>
-      </Collapse>
+      <AnimatePresence>
+         {isVisible && (
+            <Flex
```

Comment:
> Note: Direct children must each have a unique key prop so AnimatePresence can track their presence in the tree.

I'm guessing this doesn't matter because there's only one child, but this may be worth being aware of (the simple demos don't seem to include `key`, so I'm guessing it's fine in this case.)

https://www.framer.com/motion/animate-presence/##exit-animations

Reply:
Yea, I saw that, but everything seemed fine when testing. I saw their demos do not as well.